circuits/cpp/barretenberg/cpp/src/barretenberg/smt_verification/circuit/circuit.cpp

```diff
@@ -0,0 +1,124 @@
+#include "circuit.hpp"
+
+namespace smt_circuit {
+
+/**
+ * @brief Get the CircuitSchema object
+ * @details Initialize the CircuitSchmea from the binary file
+ * that contains an msgpack compatible buffer.
+ * 
+ * @param filename 
+ * @return CircuitSchema 
+ */
+CircuitSchema unpack_from_file(const std::string& filename)
+{
+    std::ifstream fin;
+    fin.open(filename, std::ios::in | std::ios::binary);
+    if (!fin.is_open()) {
+        throw std::invalid_argument("file not found");
+    }
+    if (fin.tellg() == -1) {
+        throw std::invalid_argument("something went wrong");
+    }
+
+    fin.ignore(std::numeric_limits<std::streamsize>::max()); // ohboy
+    std::streamsize fsize = fin.gcount();
+    fin.clear();
+    fin.seekg(0, std::ios_base::beg);
+    info("File size: ", fsize);
+
+    CircuitSchema cir;
+    char* encoded_data = (char*)aligned_alloc(64, static_cast<size_t>(fsize));
+    fin.read(encoded_data, fsize);
+    msgpack::unpack((const char*)encoded_data, static_cast<size_t>(fsize)).get().convert(cir);
+    return cir;
+}
+
+/**
+ * @brief Get the CircuitSchema object
+ * @details Initialize the CircuitSchmea from the msgpack compatible buffer.
+ * 
+ * @param buf 
+ * @return CircuitSchema 
+ */
+CircuitSchema unpack_from_buffer(const msgpack::sbuffer& buf)
+{
+    CircuitSchema cir;
+    msgpack::unpack(buf.data(), buf.size()).get().convert(cir);
+    return cir;
+}
+
+/**
+ * @brief Check your circuit for witness uniqness
+ * 
+ * @details Creates two Circuit objects that represent the same
+ * circuit, however you can choose which variables should be (not) equal in both cases,
+ * and also the variables that should (not) be equal at the same time.
+ *  
+ * @param circuit_info 
+ * @param s pointer to the global solver
+ * @param equal all the variables that should be equal in both circuits
```

Comment:
1. The list of names of variables which should be equal in both circuits (each is equal)
2. The list of names of variables which should not be equal in both circuits (each is not equal)
3. The list of variables, where at least one pair has to be equal
4. The list of variables, where at least one pair has to be distinct